From e686a9aeb4900658797ea748088ae6298f5ca159 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Wed, 16 Aug 2023 07:21:53 +0100
Subject: [PATCH 162/180] ld: ld-lib.exp: log failed dump.out contents for
 debugging

If we're using dump_prog in a test which fails, log the dump.out contents
to ld.log to aid debugging.

This avoids needing to ask reporters to manually run e.g. `objdump` commands
when making bug reports.

PR30722
	* ld/testsuite/lib/ld-lib.exp: Log failed dump.out contents to aid
	debugging.

Approved-by: Nick Clifton <nickc@redhat.com>
Signed-off-by: Sam James <sam@gentoo.org>
(cherry picked from commit 646657284f6b62a71a6869826e951b3def4d73a6)
---
 ld/testsuite/lib/ld-lib.exp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ld/testsuite/lib/ld-lib.exp b/ld/testsuite/lib/ld-lib.exp
index 918464069a9..aca303e0728 100644
--- a/ld/testsuite/lib/ld-lib.exp
+++ b/ld/testsuite/lib/ld-lib.exp
@@ -640,7 +640,7 @@ proc run_ld_link_tests { ldtests args } {
 		    remote_upload host "dump.out"
 
 		    if { [regexp_diff "dump.out" "$srcdir/$subdir/$dumpfile"] } then {
-			verbose "output is [file_contents "dump.out"]" 2
+			verbose -log "output is [file_contents "dump.out"]" 2
 			set failed 1
 			remote_file build delete "dump.out"
 			remote_file host delete "dump.out"
@@ -1067,7 +1067,7 @@ proc run_cc_link_tests { ldtests } {
 		    }
 
 		    if { [regexp_diff "dump.out" "$srcdir/$subdir/$dumpfile"] } then {
-			verbose "output is [file_contents "dump.out"]" 2
+			verbose -log "output is [file_contents "dump.out"]" 2
 			set failed 1
 			break
 		    }
-- 
2.43.0

